### PR TITLE
RHPAM-2617: [GSS](7.5.1) Rule logic in a RDRL file changes after importing the file as an Asset in RHPAM 7.5

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/RHPAM-2617

### CODE REVIEW

**Review by Rikkola — APPROVED** [review]

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java [hunk 1]
```diff
@@ -4306,12 +4323,18 @@ private String getDrl() {
...
+    private RuleModel getSimpleRuleModel(final String drl) {
         final RuleModel rm = new RuleModel();
         rm.setPackageName(PackageNameParser.parsePackageName(drl));
         rm.setImports(ImportsParser.parseImports(drl));
 
-        final Pattern rulePattern = Pattern.compile(".*\\s?rule\\s+(.+?)\\s+.*",
+        try {
+            parseAttributes(rm, parseDrl(drl).getAttributes());
+        } catch (Exception e) {
+            //Discard. We're unable to retrieve the rule attributes from the DRL
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java:4373]: Could you please explain why `parseDrl` so intensively catch and throw exceptions, however in final everything is ignored?

**Review by jomarko — COMMENTED** [review]
Question

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java (hunk 1 above)
Comment (manstis) [reply drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java:4373]: Well..... I copied the other `parseDrl(..)` method originally written by Mario Fusco (so I assumed he knew what he was doing..). The new method is used to retrieve the rule's _attributes_. If either parsing the DRL threw an exception or it contained errors we're unable to get the _attributes_.

Comment (jomarko) [reply drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java:4373]: @mariofusco  can we or can not we implement the new parse ethod simply as:
```java
private RuleDescr parseDrl(final String drl) {
        DrlParser drlParser = new DrlParser();
        PackageDescr packageDescr;

        packageDescr = drlParser.parse(true, drl);
        if (drlParser.hasErrors()) {
            throw new RuleModelUnmarshallingException();
        }
       
        return packageDescr.getRules().get(0);
    }
```

Comment (manstis) [reply drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java:4373]: @jomarko I've revised the method in a new commit.

**Review by jomarko — APPROVED** [review]
@manstis fine for me, but for this one Anna is final reviewer.

@dupliaka please have a look on the PR when you have a moment.

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Review by dupliaka — APPROVED** [review]
Working fine for different levels of nesting. Thanks!

**Comment by manstis** [comment]
Jenkins execute compile downstream build

**Pull request merged**